### PR TITLE
Support InternLM models

### PR DESCRIPTION
**Pull request by wangruohui**
This PR support InternLM model based on main branch. It is a follow up of #123 with rebased on to main.

Supported TRTLLM build options includes:
- [x] FP16 / BP16
- [x] TP / PP
- [x] int4/8 weight only 
- [x] int8 weight + int8 kv cache
- [x] smooth quant

Tested on both internlm-chat-7b and internlm-chat-20b.

### CODE REVIEW

File: examples/internlm/README.md [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+
+The TensorRT-LLM InternLM implementation can be found in [tensorrt_llm/models/internlm/model.py](../../tensorrt_llm/models/internlm/model.py). The TensorRT-LLM InternLM example code is located in [`examples/internlm`](./). There are three main files in that folder::
+
+ * [`build.py`](./build.py) to build the [TensorRT](https://developer.nvidia.com/tensorrt) engine(s) needed to run the InternLM model,
+ * [`run.py`](./run.py) to run the inference on an input text,
+ * [`summarize.py`](./summarize.py) to summarize the articles in the [cnn_dailymail](https://huggingface.co/datasets/cnn_dailymail) dataset using the model.
+
+## Support Matrix
+  * FP16 / BF16
+  <!-- * FP8 -->
```
Comment (nv-guomingz) [inline examples/internlm/README.md]: Please delete all comment lines in support Matrix.

File: examples/internlm/README.md [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+                    --data_type fp16 \
+                    --engine_dir ./internlm-chat-7b/trt_engines/smoothquant/1-gpu
+
+python summarize.py --test_trt_llm --test_hf \
+                    --hf_model_location ./internlm-chat-20b \
+                    --data_type fp16 \
+                    --engine_dir ./internlm-chat-20b/trt_engines/smoothquant/1-gpu
+```
+
+#### FP8 Post-Training Quantization
```
Comment (nv-guomingz) [inline examples/internlm/README.md]: No need to mention FP8 at current stage.

File: examples/internlm/README.md [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+                --use_gemm_plugin float16 \
+                --output_dir ./tmp/internlm/70B/trt_engines/fp8/2-gpu/ \
+                --remove_input_padding \
+                --enable_fp8 \
+                --fp8_kv_cache \
+                --world_size 2 \
+                --tp_size 2
+```
+-->
+#### Groupwise quantization (AWQ/GPTQ)
```
Comment (nv-guomingz) [inline examples/internlm/README.md]: Remove this part.

File: examples/llama/README.md [hunk 4]
```diff
@@ -161,7 +161,7 @@ and then export the scaling factors needed for INT8 KV cache inference.
 Example:
 
 ```bash
-python3 hf_llama_convert.py -i /llama-models/llama-7b-hf -o /llama/smooth_llama_7B/int8_kv_cache/ --calibrate-kv-cache -t fp16
+python3 hf_llama_convert.py -i llama-7b-hf -o smooth_llama_7B/int8_kv_cache/ --calibrate-kv-cache -t fp16
```
Comment (nv-guomingz) [inline examples/llama/README.md]: why we need to update llama doc?

File: docs/source/precision.md [hunk 5]
```diff
@@ -131,6 +131,7 @@ This release of TensorRT-LLM contains the following examples:
 | OPT                         | Y    | Y    | Y    | .    | .       | .     | .     | .         | .          |
 | SantaCoder                  | Y    | Y    | Y    | .    | .       | .     | .     | .         | .          |
 | StarCoder                   | Y    | Y    | Y    | .    | .       | .     | .     | .         | .          |
+| InternLM                    | Y    | Y    | Y    | ?    | Y       | Y     | Y     | ?         | ?          |
```
Comment (nv-guomingz) [inline docs/source/precision.md]: Set ? to .

File: examples/internlm/README.md [hunk 6]
```diff
@@ -0,0 +1,428 @@
+# InternLM
+
+This document shows how to build and run InternLM 7B / 20B models in TensorRT-LLM on both single GPU, single node multi-GPU and multi-node multi-GPU.
```
Comment (nv-guomingz) [inline examples/internlm/README.md:3]: InternLM ->InternLM chat. Is it more accurate?

File: examples/internlm/README.md (hunk 1 above)
Comment (wangruohui) [reply examples/internlm/README.md]: Removed.

File: examples/internlm/README.md (hunk 2 above)
Comment (wangruohui) [reply examples/internlm/README.md]: Removed.

File: examples/internlm/README.md (hunk 3 above)
Comment (wangruohui) [reply examples/internlm/README.md]: Removed.

File: examples/llama/README.md (hunk 4 above)
Comment (wangruohui) [reply examples/llama/README.md]: a mistake during commit, recovered.

File: docs/source/precision.md (hunk 5 above)
Comment (wangruohui) [reply docs/source/precision.md]: Done.

File: examples/internlm/README.md (hunk 6 above)
Comment (wangruohui) [reply examples/internlm/README.md:3]: I think we don't need too much details here. Instructions here also applies to models without chat (a.k.a. base models). They have same structures, just with different weight trained on different data and used in different scenarios. 

To make users aware of the difference, I have mentioned the differences and added detailed URLs in the next paragraph. So users following the README should found the correct one. 

There are also base models and chat models for LLama v2, but we just call them all LLama 2.

Comment (nv-guomingz) [reply examples/internlm/README.md:3]: make sense

**Comment by nv-guomingz** [comment]
Hi @wangruohui , we're going to publish lots of update to main branch this week.
Base on our internal discussion, we'll squash all features into  one commit.(e.g, [#188](https://github.com/NVIDIA/TensorRT-LLM/commit/4de32a86ae92bc49a7ec17c00ec2f2d03663c198).

Of coz, we'll add your name into the **co-author** list and credit your work on the release notes for internllm model.

Is that ok for you?

**Comment by wangruohui** [comment]
> Hi @wangruohui , we're going to publish lots of update to main branch this week. Base on our internal discussion, we'll squash all features into one commit.(e.g, [#188](https://github.com/NVIDIA/TensorRT-LLM/commit/4de32a86ae92bc49a7ec17c00ec2f2d03663c198).
> 
> Of coz, we'll add your name into the **co-author** list and credit your work on the release notes for internllm model.
> 
> Is that ok for you?

Hi @nv-guomingz ,

It's OK for me. Please move on.

**Comment by kaiyux** [comment]
Hi @wangruohui , we just pushed an update to the [main](https://github.com/NVIDIA/TensorRT-LLM/tree/main) branch, and we added you as co-author, which is also mentioned in the [announcement](https://github.com/NVIDIA/TensorRT-LLM/discussions/305#discussion-5823043).

I'm going to close this PR, please let us know if you have any questions. Thanks again for the great contribution.

**Comment by RunningLeon** [comment]
@nv-guomingz @kaiyux 

Hi, guys. Sorry to trouble you. I want to support InternLM2 models such as [internlm2-chat-7b](https://huggingface.co/internlm/internlm2-chat-7b) into your amazing project. Could you guys give me some tips on how to do it more neatly? Here are some things that should be considered:
- InternLM2 uses a merged [Wqkv](https://huggingface.co/internlm/internlm2-chat-7b/blob/4275caa205dbb8ff83930e2c1ce6bc62ec49329c/modeling_internlm2.py#L286) and GQA
- InternLM2 has different weight names compared  with llama2
- InternLM2 may use [DynamicNTKScalingRotaryEmbedding](https://huggingface.co/internlm/internlm2-chat-7b/blob/4275caa205dbb8ff83930e2c1ce6bc62ec49329c/modeling_internlm2.py#L189)

I've tried to modify [examples/llama/convert_checkpoint.py](https://github.com/NVIDIA/TensorRT-LLM/blob/main/examples/llama/convert_checkpoint.py) and it could build and run on fp16. But how could it be done more neatly? Should I create a new `examples/internlm2/convert_checkpoint.py`?

Looking forward to your kind reply.
Best Regards.

**Comment by nv-guomingz** [comment]
Hi @RunningLeon,

Thanks for your contributing on TensorRT-LLM.

It's ok for me to create the unique or dedicated internlm2 folder at current stage but we can't guarantee that it won't not be removed (merged into internlm) later.  We're actively working on new builder workflow and relateive api refactoring, the convert_checkpoint.py will be a thin wrapper in the next formal release.

Thanks,
Guoming

**Comment by ChengYouFancy** [comment]
Hi, @RunningLeon 

I'm troubled by internlm2 to tensorrt, I'v tried use llama convert, but weight names are different.
Could you share your examples/internlm2/convert_checkpoint.py code with me? And I will appreciate very much!
Looking forward to your kind reply.
Thanks

**Comment by RunningLeon** [comment]
> Hi @RunningLeon,
> 
> Thanks for your contributing on TensorRT-LLM.
> 
> It's ok for me to create the unique or dedicated internlm2 folder at current stage but we can't guarantee that it won't not be removed (merged into internlm) later. We're actively working on new builder workflow and relateive api refactoring, the convert_checkpoint.py will be a thin wrapper in the next formal release.
> 
> Thanks, Guoming

@nv-guomingz hi, thanks for your kind reply. I noticed that in the new workflow for quantization, we are using `quantize.py` which is using ammo. Is ammo open source? how do user contribute to ammo to support new models?
Thanks.

**Comment by RunningLeon** [comment]
> Hi, @RunningLeon
> 
> I'm troubled by internlm2 to tensorrt, I'v tried use llama convert, but weight names are different. Could you share your examples/internlm2/convert_checkpoint.py code with me? And I will appreciate very much! Looking forward to your kind reply. Thanks


@ChengYouFancy hi, for internlm2, you can convert it to llama style, but there's some thing you need to change.
1. Using the convert [tools](https://github.com/InternLM/InternLM/tree/main/tools) as mentioned in https://github.com/InternLM/InternLM/issues/714#issuecomment-1993355428
2.  fix the rope_scaling with dynamic as mentioned in https://github.com/NVIDIA/TensorRT-LLM/issues/1299#issuecomment-1997472337
3. you may need to change tokenizer config to skip special token

once there's a good pipeline, we'll pull a request to tensorrt-llm.

**Comment by ChengYouFancy** [comment]
> > Hi, @RunningLeon
> > I'm troubled by internlm2 to tensorrt, I'v tried use llama convert, but weight names are different. Could you share your examples/internlm2/convert_checkpoint.py code with me? And I will appreciate very much! Looking forward to your kind reply. Thanks
> 
> @ChengYouFancy hi, for internlm2, you can convert it to llama style, but there's some thing you need to change.
> 
> 1. Using the convert [tools](https://github.com/InternLM/InternLM/tree/main/tools) as mentioned in [[Feature] 是否已经支持tensorrt-llm或者计划支持？ InternLM/InternLM#714 (comment)](https://github.com/InternLM/InternLM/issues/714#issuecomment-1993355428)
> 2. fix the rope_scaling with dynamic as mentioned in [v0.8.0 long text(100k) input, output precision worse(error) #1299 (comment)](https://github.com/NVIDIA/TensorRT-LLM/issues/1299#issuecomment-1997472337)
> 3. you may need to change tokenizer config to skip special token
> 
> once there's a good pipeline, we'll pull a request to tensorrt-llm.

Thanks for your kind reply, I followed your pipeline to convert it to tensorrt successfully. It's an amazing job.
but the result of inference is incorrect, it's always "<|im_start|>" token.
Here is some situation about the model:
1. I finetuned the internlm2-chat model
2. I use the intern2 template correctly in finetune and inference
3. I did not change tokenizer config to skip special token as you suggest, I'm not sure what should I change

Do you have some idea about the inference result?
Thanks for your help again!

**Comment by RunningLeon** [comment]
@ChengYouFancy hi, did you fix the rope in cpp and rebuild trtllm?

**Comment by ChengYouFancy** [comment]
> @ChengYouFancy hi, did you fix the rope in cpp and rebuild trtllm?

I followed the pipeline:
1. use convert2llama.py in InternLm repo to convert internlm2 to llama
   
3. I fixed the rope in cpp
4. use convert_checkpoint.py in TensorRT/example/llama to convert the model of  first step
`CUDA_VISIBLE_DEVICES=2 python3 convert_checkpoint.py --model_dir /workspace/model/internlm-7b-chat-all-sft-convert-llama/ \
                --dtype float16 \
                --output_dir /workspace/model/internlm-7b-chat-all-sft-trt_engines`
5. use trtllm-build to build trtllm
`CUDA_VISIBLE_DEVICES=2 trtllm-build --checkpoint_dir /workspace/model/internlm-7b-chat-all-sft-trt_engines \
            --output_dir /workspace/model/internlm-7b-chat-all-sft-tensorrt \
            --gemm_plugin float16 \
            --remove_input_padding enable \
            --max_batch_size 8 \
            --max_output_len 8192`
6. test the inference based on run.py
    the inference is incorrect

**Comment by RunningLeon** [comment]
@ChengYouFancy hi, you may need to check the input prompt to see if it has the correct chat template. You can add a template in here
https://github.com/NVIDIA/TensorRT-LLM/blob/850b6fa1e710d25769f2b560d897d2bd424a645e/examples/utils.py#L41

```
    'LlamaForCausalLM':
    "<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\n{input_text}<|im_end|>\n<|im_start|>assistant\n",

**Comment by ChengYouFancy** [comment]
> @ChengYouFancy hi, you may need to check the input prompt to see if it has the correct chat template. You can add a template in here
> 
> https://github.com/NVIDIA/TensorRT-LLM/blob/850b6fa1e710d25769f2b560d897d2bd424a645e/examples/utils.py#L41
> 
> ```
>     'LlamaForCausalLM':
>     "<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\n{input_text}<|im_end|>\n<|im_start|>assistant\n",
> ```

input text looks like is correct
`Input [Text 0]: " <s><|im_start|>  system
You are an AI assistant whose name is InternLM (书生·浦语).
- InternLM (书生·浦语) is a conversational language model that is developed by Shanghai AI Laboratory (上海人工智能实验室). It is designed to be helpful, honest, and harmless.
- InternLM (书生·浦语) can understand and communicate fluently in the language chosen by the user such as English and 中文.<|im_end|>
<|im_start|>  user
中国的首都是哪里<|im_end|>
<|im_start|> assistant
"
Output [Text 0 Beam 0]: " <|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|><|im_start|>"`

**Comment by RunningLeon** [comment]
@ChengYouFancy Have you rebuild  and reinstall trtllm ? 
```
# Build TensorRT-LLM for Ampere.
python3 ./scripts/build_wheel.py --cuda_architectures "80-real;86-real" --trt_root /usr/local/tensorrt
```

**Comment by ChengYouFancy** [comment]
> @ChengYouFancy Have you rebuild and reinstall trtllm ?
> 
> ```
> # Build TensorRT-LLM for Ampere.
> python3 ./scripts/build_wheel.py --cuda_architectures "80-real;86-real" --trt_root /usr/local/tensorrt
> ```

I haven't, should I do it after I fixed the rope in cpp?

**Comment by RunningLeon** [comment]
> > @ChengYouFancy Have you rebuild and reinstall trtllm ?
> > ```
> > # Build TensorRT-LLM for Ampere.
> > python3 ./scripts/build_wheel.py --cuda_architectures "80-real;86-real" --trt_root /usr/local/tensorrt
> > ```
> 
> I haven't, should I do it after I fixed the rope in cpp?

Yes. You should rebuild and reinstall after fixing cpp

**Comment by ChengYouFancy** [comment]
> cpp

Thanks a lot, I will try it

**Comment by ChengYouFancy** [comment]
> > > @ChengYouFancy Have you rebuild and reinstall trtllm ?
> > > ```
> > > # Build TensorRT-LLM for Ampere.
> > > python3 ./scripts/build_wheel.py --cuda_architectures "80-real;86-real" --trt_root /usr/local/tensorrt
> > > ```
> > 
> > 
> > I haven't, should I do it after I fixed the rope in cpp?
> 
> Yes. You should rebuild and reinstall after fixing cpp

I'v tried to rebuild tensorrtllm, but cmake failed. and I could not search any similar issue to help me
error info:
`subprocess.CalledProcessError: Command 'cmake -DCMAKE_BUILD_TYPE="Release" -DBUILD_PYT="ON" -DBUILD_PYBIND="ON" -DNVTX_DISABLE="ON" "-DCMAKE_CUDA_ARCHITECTURES=80-real;86-real" -DTRT_LIB_DIR=/usr/local/tensorrt/targets/x86_64-linux-gnu/lib -DTRT_INCLUDE_DIR=/usr/local/tensorrt/include  -S "/workspace/code/TensorRT-LLM/cpp"' returned non-zero exit status 1.`
image

**Comment by nv-guomingz** [comment]
> > Hi @RunningLeon,
> > Thanks for your contributing on TensorRT-LLM.
> > It's ok for me to create the unique or dedicated internlm2 folder at current stage but we can't guarantee that it won't not be removed (merged into internlm) later. We're actively working on new builder workflow and relateive api refactoring, the convert_checkpoint.py will be a thin wrapper in the next formal release.
> > Thanks, Guoming
> 
> @nv-guomingz hi, thanks for your kind reply. I noticed that in the new workflow for quantization, we are using `quantize.py` which is using ammo. Is ammo open source? how do user contribute to ammo to support new models? Thanks.

Hi @RunningLeon 

AMMO is not open-sourced yet and I don't we have plan to open source it soon.
 
>how do user contribute to ammo to support new models?

Currently, the typical workflow is we add a new model supporting on high precision and AMMO team will responsible for quantizing the model.

**Comment by RunningLeon** [comment]
> > > Hi @RunningLeon,
> > > Thanks for your contributing on TensorRT-LLM.
> > > It's ok for me to create the unique or dedicated internlm2 folder at current stage but we can't guarantee that it won't not be removed (merged into internlm) later. We're actively working on new builder workflow and relateive api refactoring, the convert_checkpoint.py will be a thin wrapper in the next formal release.
> > > Thanks, Guoming
> > 
> > 
> > @nv-guomingz hi, thanks for your kind reply. I noticed that in the new workflow for quantization, we are using `quantize.py` which is using ammo. Is ammo open source? how do user contribute to ammo to support new models? Thanks.
> 
> Hi @RunningLeon
> 
> AMMO is not open-sourced yet and I don't we have plan to open source it soon.
> 
> > how do user contribute to ammo to support new models?
> 
> Currently, the typical workflow is we add a new model supporting on high precision and AMMO team will responsible for quantizing the model.

@nv-guomingz hi, Got it. Thanks for your quick reply.

**Comment by ChengYouFancy** [comment]
> > > @ChengYouFancy Have you rebuild and reinstall trtllm ?
> > > ```
> > > # Build TensorRT-LLM for Ampere.
> > > python3 ./scripts/build_wheel.py --cuda_architectures "80-real;86-real" --trt_root /usr/local/tensorrt
> > > ```
> > 
> > 
> > I haven't, should I do it after I fixed the rope in cpp?
> 
> Yes. You should rebuild and reinstall after fixing cpp

Thank you very much, after rebuild and reinstall trtllm, I've convert internlm2 to tensorrt_llm successfully and the inference result is correct finally
Best Regard

**Comment by RunningLeon** [comment]
@nv-guomingz hi, could you kindly review this PR https://github.com/NVIDIA/TensorRT-LLM/pull/1392 if you have time? Thanks in advance.

**Comment by nv-guomingz** [comment]
> @nv-guomingz hi, could you kindly review this PR https://github.com/NVIDIA/TensorRT-LLM/pull/1392 if you have time? Thanks in advance.

Hi @RunningLeon , thanks for your great contribution to TRT-LLM project.
We'll review your MR soon and merge your code into our mainstream once it verified.
And then we'll cite your contribution into our release notes.

cc  @juney-nvidia @kaiyux @byshiue for vis

**Comment by icoderzqliu** [comment]
> > Hi, @RunningLeon
> > I'm troubled by internlm2 to tensorrt, I'v tried use llama convert, but weight names are different. Could you share your examples/internlm2/convert_checkpoint.py code with me? And I will appreciate very much! Looking forward to your kind reply. Thanks
> 
> @ChengYouFancy hi, for internlm2, you can convert it to llama style, but there's some thing you need to change.
> 
> 1. Using the convert [tools](https://github.com/InternLM/InternLM/tree/main/tools) as mentioned in [[Feature] 是否已经支持tensorrt-llm或者计划支持？ InternLM/InternLM#714 (comment)](https://github.com/InternLM/InternLM/issues/714#issuecomment-1993355428)
> 2. fix the rope_scaling with dynamic as mentioned in [v0.8.0 long text(100k) input, output precision worse(error) #1299 (comment)](https://github.com/NVIDIA/TensorRT-LLM/issues/1299#issuecomment-1997472337)
> 3. you may need to change tokenizer config to skip special token
> 
> once there's a good pipeline, we'll pull a request to tensorrt-llm.

Hello, I was doing inference after converting internlm2 to llama and got an error:
```
Loading checkpoint shards:   0%|                                                                                                                | 0/21 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/data3/zqliu/program/LLaMA-Factory/law/infer_laws.py", line 75, in <module>
    model = AutoModelForCausalLM.from_pretrained(model_path,
  File "/home/icoderzqliu/miniconda3/envs/lf/lib/python3.10/site-packages/transformers/models/auto/auto_factory.py", line 566, in from_pretrained
    return model_class.from_pretrained(
  File "/home/icoderzqliu/miniconda3/envs/lf/lib/python3.10/site-packages/transformers/modeling_utils.py", line 3706, in from_pretrained
    ) = cls._load_pretrained_model(
  File "/home/icoderzqliu/miniconda3/envs/lf/lib/python3.10/site-packages/transformers/modeling_utils.py", line 4091, in _load_pretrained_model
    state_dict = load_state_dict(shard_file)
  File "/home/icoderzqliu/miniconda3/envs/lf/lib/python3.10/site-packages/transformers/modeling_utils.py", line 503, in load_state_dict
    with safe_open(checkpoint_file, framework="pt") as f:
safetensors_rust.SafetensorError: Error while deserializing header: HeaderTooLarge
```
Can the converted model be loaded directly using AutoModelForCausalLM.from_pretrained? Or should it be directly converted into tensorrt form?

**Comment by RunningLeon** [comment]
@icoderzqliu hi, the converted llama-style internlm2 can be loaded and run with transformers.  Not met your issue. Are you using internlm2-chat-20b?  BTW, you can try pr #1392, no need to convert to llama style.

**Comment by icoderzqliu** [comment]
> @icoderzqliu hi, the converted llama-style internlm2 can be loaded and run with transformers. Not met your issue. Are you using internlm2-chat-20b? BTW, you can try pr #1392, no need to convert to llama style.

@RunningLeon hi, thank you for your quick reply. Yes, I use internlm2-chat-20b. I will try to use this pr https://github.com/NVIDIA/TensorRT-LLM/pull/1392. It seems that the source code needs to be changed. Is there any more detailed steps that I can follow?

**Comment by ChengYouFancy** [comment]
@RunningLeon hi，we use lora finetuned the internlm2 model. Now we can convert the base model to llama, but not lora part, we tried to change the code of InternLM/tools/convert2llame.py to transfer lora to llama style, but did not work. Is there any other tools could work for lora?
tips: we can't export base model and lora to one single model, because we want to use https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples/llama#run-llama-with-several-lora-checkpoints this feature. Thus, if we can transfer internlm2 lora to llama style, then we can use examples/hf_lora_convert.py to build trt-llm

**Comment by RunningLeon** [comment]
@ChengYouFancy hi, sorry for the late reply due to the vacation. Maybe you can open an issue to https://github.com/InternLM/InternLM since the convert script is provided by them.

**Pull request closed without merging**